### DISCUSSION TITLE
How to get push notification stats in Firebase Console

### DISCUSSION THREAD

**Discussion by Heshyo** [ionic-team/capacitor#6978] [Q&A]
In Firebase Console Messaging Reports, I see a graph with the number of notifications sent. I trust that the values shown are correct. But for Received, Impressions and Open count, it's always 0.

Is this to be expected? Isn't the plugin supposed to send necessary data to Firebase to track those?

FYI, push notifications work well on both Android and iOS (I use https://github.com/capacitor-community/fcm to also use Firebase for iOS). I send my notifications from my backend (I'm not using any Firebase SDK).

**Comment by iamkinetic** [comment]
@Heshyo I'm looking for the exact same thing.  Did you find what was missing?

**Comment by Heshyo** [comment]
@iamkinetic No I didn't really look into it. It's not a blocking issue for me as I've already been tracking the number of opened notifications even before using Firebase, but i expected to get that kind of data and more "for free" using Firebase. Someone more familiar with Firebase Notification SDK might be able to help.

**Reply by yemitula** [reply]
I have a similar issue and I hope someone helps. However, how are you currently tracking the number of opened notifications?

**Reply by Heshyo** [reply]
I listen to the `pushNotificationActionPerformed` event and sends the ID corresponding to the notification to my own back end (my notification contains some extra data, one being an ID).

```
await PushNotifications.addListener('pushNotificationActionPerformed', actionPerformed => {
  // This is called if the app is closed or in the background, and the user clicks the notification.
  this.registerOpenNotificationToBackend(actionPerformed.notification);
});
```

**Reply by yemitula** [reply]
Great! Thanks for sharing
